training/tf/parse.py

```diff
@@ -52,7 +62,36 @@ def remap_vertex(vertex, symmetry):
     return y * 19 + x
 
 class ChunkParser:
-    def __init__(self, chunks, workers=None):
+    def __init__(self, chunkdatagen, shuffle_size=1, sample=1, buffer_size=1, workers=None):
+        """
+            Read data and yield batches of raw tensors.
+
+            'chunkdatagen' is a generator yeilding chunkdata
+            'shuffle_size' is the size of the shuffle buffer.
+            'sample' is the rate to down-sample.
+            'workers' is the number of child workers to use.
+
+            The data is represented in a number of formats through this dataflow
+            pipeline. In order, they are:
+
+            chunk: The name of a file containing chunkdata
+
+            chunkdata: type Bytes. Either mutiple records of v1 format, or multiple records
+            of v2 format.
+
+            v1: The original text format describing a move. 19 lines long. VERY slow
+            to decode. Typically around 2500 bytes long. Used only for backward
```

Comment:
should this be 27568 (4 + 4 * 393 + 4 * 6498)?

Reply:
Not quite sure where that number came from? The minimum size is 2200 (everything zero). The max size depends on the number of non-zero probabilities. The average I measured was about 2500 bytes per move?